src/logic/community/broadcaster.ts

```diff
@@ -222,6 +242,85 @@ export function createCommunityBroadcasterComponent(
     )
   }
 
+  /**
+   * Gets online community members with pagination support
+   * @param {string} communityId - The ID of the community
+   * @param {string[]} onlineUsers - Array of online user addresses to filter by
+   * @param {string[]} excludedAddresses - Optional array of addresses to exclude
+   * @returns {Promise<string[]>} Array of online member addresses
+   */
+  async function getOnlineCommunityMembersAddresses(
+    communityId: string,
+    onlineUsers: string[],
+    excludedAddresses?: string[]
+  ): Promise<string[]> {
+    const onlineMemberAddresses: string[] = []
+    let offset = 0
+    let hasMore = true
+
+    while (hasMore) {
+      const batch = await communitiesDb.getCommunityMembers(communityId, {
+        pagination: { limit: MEMBER_FETCH_BATCH_SIZE, offset },
+        filterByMembers: onlineUsers,
+        excludedAddresses
+      })
+
+      if (batch.length === 0) break
+
+      batch.forEach(({ memberAddress }) => {
+        onlineMemberAddresses.push(memberAddress)
+      })
+
+      offset += MEMBER_FETCH_BATCH_SIZE
+      hasMore = batch.length === MEMBER_FETCH_BATCH_SIZE
+    }
+
+    return onlineMemberAddresses
+  }
+
+  /**
+   * Broadcasts voice chat started event to online community members
+   * @param {BroadcastableEvent} event - The event to broadcast
+   * @param {BroadcastOptions} options - Optional broadcast options (e.g., excludeAddresses)
+   */
+  async function broadcastVoiceChatStarted(event: BroadcastableEvent, options?: BroadcastOptions): Promise<void> {
+    const { metadata } = event as CommunityVoiceChatStartedEventReducedMetadata
+    const { communityId } = metadata
+
+    const onlineSubscribers = subscribersContext.getSubscribersAddresses()
+
+    if (onlineSubscribers.length === 0) {
+      return
+    }
+
+    // Get online members from the community, excluding specified addresses at database level
+    const excludedAddresses = options?.excludeAddresses?.map((addr) => addr.toLowerCase())
+    const onlineMemberAddresses = await getOnlineCommunityMembersAddresses(
+      communityId,
+      onlineSubscribers,
+      excludedAddresses
+    )
+
+    if (onlineMemberAddresses.length === 0) {
+      return
+    }
+
+    const memberBatches = createMemberBatches(onlineMemberAddresses)
+
+    await sns.publishMessages(
+      memberBatches.map((batch, i) => ({
+        ...event,
+        key: `${event.key}-batch-${i + 1}`,
```

Comment:
NIT:
The key can be a generic one, since the event will arrive once per user address.

DOUBT:
What's `event.key` here? asking because if it is something like `<communityId>`, all voice chats starting will have same same key, meaning that only the first voice chat will be notified, afterwards notification would detect that there is already a notification for that key and that address and exclude them on retrieval.

Even if we add the `batch-<number>`, that index will be the same.

I think we already have a unique event.keys at this point but since I am not sure I leave this comment 👼